pandas_gbq/gbq.py

```diff
@@ -603,106 +458,56 @@ def delete_and_recreate_table(self, dataset_id, table_id, table_schema):
-    col_names = [str(field['name']) for field in fields]
-    col_dtypes = [dtype_map.get(field['type'], object) for field in fields]
-    page_array = np.zeros((len(rows),), dtype=lzip(col_names, col_dtypes))
-    for row_num, raw_row in enumerate(rows):
-        entries = raw_row.get('f', [])
-        for col_num, field_type in enumerate(col_types):
-            field_value = _parse_entry(entries[col_num].get('v', ''),
-                                       field_type)
-            page_array[row_num][col_num] = field_value
-
-    return DataFrame(page_array, columns=col_names)
-
-
-def _parse_entry(field_value, field_type):
-    if field_value is None or field_value == 'null':
-        return None
-    if field_type == 'INTEGER':
-        return int(field_value)
-    elif field_type == 'FLOAT':
-        return float(field_value)
-    elif field_type == 'TIMESTAMP':
-        timestamp = datetime.utcfromtimestamp(float(field_value))
-        return np.datetime64(timestamp)
-    elif field_type == 'BOOLEAN':
-        return field_value == 'true'
-    return field_value
-
-
-def read_gbq(query, project_id=None, index_col=None, col_order=None,
-             reauth=False, verbose=True, private_key=None, dialect='legacy',
-             **kwargs):
-    r"""Load data from Google BigQuery.
+def read_gbq(query, project_id=None, index_col=None, col_order=None, verbose=True,
+             private_key=None, dialect='legacy', configuration=None, **kwargs):
+    r"""Load data from Google BigQuery using google-cloud-python
 
     The main method a user calls to execute a Query in Google BigQuery
     and read results into a pandas DataFrame.
 
-    Google BigQuery API Client Library v2 for Python is used.
+    The Google Cloud library is used.
     Documentation is available `here
-    <https://developers.google.com/api-client-library/python/apis/bigquery/v2>`__
-
-    Authentication to the Google BigQuery service is via OAuth 2.0.
-
-    - If "private_key" is not provided:
+    <https://googlecloudplatform.github.io/google-cloud-python/stable/>`__
 
-      By default "application default credentials" are used.
-
-      If default application credentials are not found or are restrictive,
-      user account credentials are used. In this case, you will be asked to
-      grant permissions for product name 'pandas GBQ'.
-
-    - If "private_key" is provided:
-
-      Service account credentials will be used to authenticate.
+    Authentication via Google Cloud can be performed a number of ways, see:
+    <https://googlecloudplatform.github.io/google-cloud-python/stable/google-cloud-auth.html>
+    The easiest is to download a service account JSON keyfile and point to it using 
+    an environment variable:
+    `$ export GOOGLE_APPLICATION_CREDENTIALS="/path/to/keyfile.json"`
 
     Parameters
     ----------
     query : str
         SQL-Like Query to return data values
-    project_id : str
+    project_id : str (optional)
```

Comment:
how is this optional?

Reply:
Hm, doc still says that the project will be inferred from environment (https://googlecloudplatform.github.io/google-cloud-python/stable/bigquery/client), but I don't think it does anymore in the latest version. Thus, project_id is now required again, which is incredibly annoying as most of the time, you're probably ok with the query being associated with the same project. Thoughts on whether we should allow the user to specify a default project env variable or other method? (`~/.bigqueryrc` looks like it can hold your default project_id, but I don't know if that's deprecated and/or a command-line only implementation: https://cloud.google.com/bigquery/bq-command-line-tool)

@MaximilianR @jreback